src/ElectricityMarketData.jl

```diff
@@ -1,5 +1,64 @@
 module ElectricityMarketData
 
-# Write your package code here.
+using Dates
+using Logging
+using Tables
 
+export ElectricityMarket, list_markets, available_time_series, get_rt_lmp_raw_data, rt_lmp_data
+
+"""
+    ElectricityMarket
+
+Abstract type representing an electricity market.
+"""
+abstract type ElectricityMarket end
+
+"""
+    list_markets() :: Vector{ElectricityMarket}
+
+Return a vector of all available electricity markets.
+"""
+function list_markets() :: Vector{Symbol}
+    return Vector{Symbol}()
+end
+
+"""
+    available_time_series(market::ElectricityMarket) :: Vector{NamedTuple}
+
+Return Vector of `NamedTuple` of available time series for the given `market`.
+
+Ex:
+```
+[
+    (name="RT-load", unit="MW", resolution=Hour(1), first_date=DateTime("2021-01-01T00:00:00"), method=rt_load_data, description="Real-time load data"),
+    (name="RT-LMP", unit="MWh", resolution=Hour(1), first_date=DateTime("2021-01-01T00:00:00"), method=rt_lmp_data, description="Real-time Locational Marginal Price data"),
+]
+```
+"""
+function available_time_series(market::ElectricityMarket)
+    @warn "No time series registered for $(market)"
+    return Vector{NamedTuple}()
 end
+
+"""
+    get_rt_lmp_raw_data(market::ElectricityMarket, start_date::DateTime, end_date::DateTime; folder::AbstractString=mktempdir(), parser::Function=(args...) -> nothing)
+
+Download raw data for Real-Time (RT) Locational Marginal Price (LMP) for the given `market` and `start_date` to `end_date` and save it in `folder`.
+Parse the data using `parser` if provided.
+"""
+function get_rt_lmp_raw_data(market::ElectricityMarket, start_date::DateTime, end_date::DateTime; folder::AbstractString="", parser::Function=(args...) -> nothing)
```

Comment:
maybe ZonedDateTime instead of DateTime?

Reply:
My fear is that the user would have to search to find out what timezone is related to each market.

I agree that it is better with  ZonedDateTime since I might want to limit my fetch based on a external time event.

Perhaps we can create a default timezone in case the user doesn't want to search?

Reply:
made the default interface be zonedatetime

Reply:
feel free reopen if you have thoughts on my treatment of transforming datetime to zonedatetime